javascripts/2048.js

```diff
@@ -32,21 +32,57 @@ function moveTile(tile, direction) {
 }
 
 
-// board = new Board([]);
 var Board = function(boardArray) { // board constructor
   this.board = boardArray;
-}
+};
+
+board = new Board([
+  [2,     2,   0,   0], // [2,   4,  16, 512],
+  [4,     8,   4,   2], // [2,   8,  32, 256],
+  [16,   32, 128,  64], // [0,   4, 128, 128],
+  [512, 256, 128,  32] //  [0,   2,  64,  32]
+]);
 
-Board.protoype.update(direction) {
+console.log(board.board);
+
+Board.prototype.move = function(direction) {
   // this is the movement function
-  // var reorientedBoard = this.split();
-  // this.condense();
-}
+  var reorientedBoard = this.split(direction);
+  console.log(reorientedBoard);
+  // var condensedBoard = reorientedBoard.forEach(this.condense); // not 100% certain about this syntax
+};
 
-// board.split()
-Board.prototype.split() {
-  // splits the board into arrays based on direction
-}
+// board.reorient() reorients the board into arrays based on direction
+Board.prototype.reorient = function(direction) {
+  var method;
+
+  if (direction == "left" || direction == "right")
+    method = "leftReorient";
+  else // "up" || "down"
+    method = "downReorient";
+
+  return this[method].call(this); // execute the method in the current context
+};
+
+Board.prototype.leftReorient = function() {
+  return this.board; // or do we want to modify the board in place?
+};
+
+Board.prototype.downReorient = function() {
```

Comment:
can we have these function names also refer to the right and up directions? Something like `leftRightReorient` and `downUpReorient`? I know it's messy, but I got a bit confused at first when looking at these functions.